hazelcast/src/main/java/com/hazelcast/nio/serialization/PortableWriter.java

```diff
@@ -124,7 +132,54 @@
      * @param classId   class ID of related portable class
      * @throws IOException in case of any exceptional case
      */
-    void writeNullPortable(String fieldName, int factoryId, int classId) throws IOException;
+    void writeNullPortable(@Nonnull String fieldName, int factoryId, int classId) throws IOException;
+
+    /**
+     * Writes a decimal which is arbitrary precision and scale floating-point number
```

Comment:
nit: trailing dot is missing here and in below
```suggestion
     * Writes a decimal which is arbitrary precision and scale floating-point number.
```